### PR TITLE
Reorganize users views into separate version files

### PR DESCRIPTION
**Pull request by reinbach**
Including the tests

See #1071

### CODE REVIEW

File: EquiTrack/users/tests/test_views_v3.py [hunk 1]
```diff
@@ -3,11 +3,14 @@
...
 from django.core.urlresolvers import reverse
 from rest_framework import status
 
-from EquiTrack.factories import UserFactory
+from EquiTrack.factories import GroupFactory, UserFactory
 from EquiTrack.tests.mixins import APITenantTestCase
+from users.serializers_v3 import AP_ALLOWED_COUNTRIES
 
 
 class TestCountryView(APITenantTestCase):
```
Comment (caktus-philip) [inline EquiTrack/users/tests/test_views_v3.py]: IMHO this test belongs in the tests file that corresponds to the views file in which implementation resides (v1 or v2 if you decide to move it there).

File: EquiTrack/users/views_v2.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+from users import views as v1
+
+
+class UserAuthAPIView(v1.UserAuthAPIView):
+    """Stub for UserAuthAPIView"""
+
```
Comment (caktus-philip) [inline EquiTrack/users/views_v2.py]: I didn't know you could define a class like this! I thought you always needed at least a `pass` statement. Thanks for the education.

**Comment by caktus-philip** [comment]
Not sure why CircleCI failed. All tests pass for me and flake8 runs clean.

**Comment by reinbach** [comment]
The failing test is due to the end date used in t2f tests, end of year. It has been fixed in other branches.

File: EquiTrack/users/views.py [hunk 3]
```diff
@@ -73,14 +73,14 @@ def post(self, request, format=None):
             return Response(status=status.HTTP_204_NO_CONTENT)
 
 
-class UsersView(ListAPIView):
+class StaffUsersView(ListAPIView):
```
Comment (domdinicola) [inline EquiTrack/users/views.py]: what's going on here?

Comment (reinbach) [reply EquiTrack/users/views.py]: Gave the class a more accurate name, as this view returns only staff users.

Comment (domdinicola) [reply EquiTrack/users/views.py]: mmm it seems you changed also the permissions no?

Comment (reinbach) [reply EquiTrack/users/views.py]: Yes, as I think we wanted to limit the list of staff users to superusers and staff and not any user logged in

**Pull request merged**